core/src/main/java/com/hivemq/testcontainer/core/GradleHiveMQExtensionSupplier.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 HiveMQ and the HiveMQ Community
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.testcontainer.core;
+
+import net.lingala.zip4j.ZipFile;
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.shaded.com.google.common.io.Files;
+import org.testcontainers.shaded.org.apache.commons.lang.SystemUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class automates the process of packaging a HiveMQ extension from a gradle project.
+ * It uses ./gradlew on MacOS and Linux and uses gradlew.bat on Windows to execute the gradle task.
+ *
+ * @author Yannick Weber
+ * @since 1.3.0
+ */
+public class GradleHiveMQExtensionSupplier implements Supplier<File> {
+
+    private static final @NotNull Pattern ROOT_PROJECT_PATTERN = Pattern.compile(".*'(.*)'");
+    private static final @NotNull String PROPERTY_REGEX = "(.*): (.*)";
+
+    private static final @NotNull String BUILD_STARTED =
+            "=================================================================\n" +
+                    "===   Embedded Gradle build started: %s   ===\n" +
+                    "=================================================================\n";
+
+    private static final @NotNull String BUILD_STOPPED =
+            "=================================================================\n" +
+                    "===   Embedded Gradle build stopped: %s   ===\n" +
+                    "=================================================================";
+
+    private static final @NotNull String TASK = "hivemqExtensionZip";
+    private static final @NotNull String GRADLE_WRAPPER_BASH_COMMAND = "./gradlew";
+
+    private final @NotNull String gradleBuild;
+    private boolean quiet = false;
+
+    /**
+     * This {@link Supplier} can be used if the current gradle project is the HiveMQ Extension to supply.
+     * It uses the build.gradle file and the gradle wrapper of the current gradle project.
+     *
+     * @return a {@link GradleHiveMQExtensionSupplier} for the current gradle project
+     * @since 1.3.0
+     */
+    public static @NotNull GradleHiveMQExtensionSupplier direct() {
+
+        return new GradleHiveMQExtensionSupplier("build.gradle");
```

Comment:
Imho it makes more sense that to pass the root directory of the project as argument because this would be agnostic to the groovy and kotlin dsl